drivers/mpsl/clock_control/CMakeLists.txt

```diff
@@ -5,4 +5,7 @@
 #
 
 zephyr_library()
-zephyr_library_sources(clock_control_mpsl.c)
```

Comment:
what about the calls to `z_nrf_clock_control_get_onoff` and `z_nrf_clock_control_lf_on` ?
Those are implemented in `clock_control_mpsl.c` but not available in `nrfx_clock_mpsl.c`.

They are called from upstream doe, as example:
https://github.com/zephyrproject-rtos/zephyr/blob/c54a511d26d0127f60ef6f2fbee6b3757defe627/drivers/timer/nrf_rtc_timer.c#L215

Reply:
see PR description.

changing the layer which is mpsl dependent. Previously it was the clock control driver. Now clock control driver is using `nrfx_clock` driver from nrfx so clock control implementation is not hw dependent but relies on nrfx driver API. This implements nrfx_clock API and zephyr clock control driver will be used with mpsl.

Reply:
Guess this should be removed when introducing this fix:
https://github.com/nrfconnect/sdk-nrf/blob/208302083ec5591fad6ede4a46ecff36db8c52a5/drivers/mpsl/clock_control/Kconfig#L7-L12

as it forces clock control in mpsl, which is no longer available.
Currently running some local tests on upmerge PR with this PR, and referred Kconfig code removed.

Reply:
part of the upmerge.